tests/test_exhaustive_layouts.cpp

```diff
@@ -391,9 +391,11 @@ TYPED_TEST(TestLayoutConversion, implicit_conversion) {
   #endif
     map1 = typename TestFixture::map_1_t(map2);
 
-  for(size_t r=0; r != this->exts1.rank(); ++r) {
-    ASSERT_EQ(map1.extents().extent(r), map2.extents().extent(r));
-    ASSERT_EQ(map1.stride(r), map2.stride(r));
+  if constexpr (TestFixture::exts_1_t::rank() > 0) {
```

Comment:
We still test this with C++14

Reply:
Given my other comment, I would just protect the condition with whether C++20 is on.

Reply:
Works for me.  Will update.

Reply:
Updated.  May not be the best-looking `#if`s...

Reply:
This code _should_ be fine even if `MDSPAN_HAS_CXX_20` defined, because the constraint on `stride(r)` won't come into play, no?  I'd rather have fewer macros if we can avoid it.

Reply:
This check is for the case when `exts_1_t::rank() == 0` in the C++20 mode, in which case we have the constraint and it'll result in a compilation error without the `if constexpr`.
Well, that being said, I think we should only protect the line that calls `stride()`.  I'll fix that.